core/src/main/scala/za/co/absa/spline/harvester/ComponentCreatorFactory.scala

```diff
@@ -18,21 +18,19 @@ package za.co.absa.spline.harvester
 
 import java.util.concurrent.atomic.AtomicInteger
 
-import org.apache.spark.sql.catalyst.expressions.{ExprId, Attribute => SparkAttribute}
 import za.co.absa.commons.lang.CachingConverter
-import za.co.absa.spline.harvester.converter.{AttributeConverter, DataConverter, DataTypeConverter, ExpressionConverter, OperationParamsConverter}
+import za.co.absa.spline.harvester.converter._
 
 class ComponentCreatorFactory {
   val dataConverter = new DataConverter
   val dataTypeConverter = new DataTypeConverter with CachingConverter
-  val attributeConverter = new AttributeConverter(dataTypeConverter) with CachingConverter {
-    override protected def keyOf(attribute: SparkAttribute): ExprId = attribute.exprId
-  }
-
-  val expressionConverter = new ExpressionConverter(dataConverter, dataTypeConverter, attributeConverter)
+  val inputAttributeConverter = new InputAttributeConverter(dataTypeConverter)
+  val expressionConverter = new GenericExpressionConverter(dataConverter, dataTypeConverter)
   val operationParamsConverter = new OperationParamsConverter(dataConverter, expressionConverter)
 
   private[this] val lastId = new AtomicInteger(0)
 
   def nextId: Int = lastId.getAndIncrement()
+
+  def expressionConverters():List[ExpressionConverter] = inputAttributeConverter :: expressionConverter :: Nil
```

Comment:
```suggestion
  def expressionConverters():List[ExpressionConverter] = Seq(inputAttributeConverter, expressionConverter)
```

Operator `::` while still being useful for extracting and pattern-matching, as a way of composing lists however is considered deprecated since about Scala 2.8.
Respective `apply()` functions or `:+`, `:+` etc methods are usually more readable.